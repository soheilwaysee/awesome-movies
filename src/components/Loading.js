import React from "react";
import styles from "../styles/components/Loading.module.css";
import Icon from "./Icon";

const Loading = ({ show }) => {
  return show ? (
    <div id="loading" className={styles.loading}>
      <Icon name="spinner" />
    </div>
  ) : null;
};

export default Loading;
