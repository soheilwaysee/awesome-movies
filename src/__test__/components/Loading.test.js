import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Loading from '../../components/Loading';

afterEach(cleanup);

it('renders Loading', async () => {
  const { container, rerender } = render(<Loading />);
  expect(container.firstChild).toBeNull();
  rerender(<Loading show />);
  expect(container.firstChild).not.toBeNull();
  expect(container.firstChild).toMatchSnapshot();
});
