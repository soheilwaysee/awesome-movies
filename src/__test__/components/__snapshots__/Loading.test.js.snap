// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading 1`] = `
<div
  class="loading"
  id="loading"
>
  <i
    class="icon icon-spin5 animate-spin"
  >
    î ¸
  </i>
</div>
`;
